atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+                    if 'energ' in observable_name:  # TODO check for energies or energy when matching name
+                        if hasattr(Energy, observable_label):
+                            data['energy'][observable_label] = dict(value=val)
+                        else:
+                            data_h5md['x_h5md_energy_contributions'].append(EnergyEntry(kind=key, value=val))
+                    else:
+                        if hasattr(BaseCalculation, observable_label):
+                            data[observable_label] = val
+                        else:
+                            unit = None
+                            if hasattr(val, 'units'):
+                                unit = val.units
+                                val = val.magnitude
+                            data_h5md['x_h5md_custom_calculations'].append(CalcEntry(kind=key, value=val, unit=unit))
+
+            self.parse_thermodynamics_step(data)
+            sec_calc = sec_run.calculation[-1]
+
+            if sec_calc.step != step:  # TODO check this comparison
+                sec_calc = sec_run.m_create(Calculation)
+                sec_calc.step = int(step)
+                sec_calc.time = data['time']
+            for calc_entry in data_h5md['x_h5md_custom_calculations']:
+                sec_calc.x_h5md_custom_calculations.append(calc_entry)
+            sec_energy = sec_calc.energy
+            if not sec_energy:
+                sec_energy = sec_calc.m_create(Energy)
+            for energy_entry in data_h5md['x_h5md_energy_contributions']:
+                sec_energy.x_h5md_energy_contributions.append(energy_entry)
+
+    def parse_system(self):
+        sec_run = self.archive.run[-1]
+
+        system_info = self._system_info.get('system')
+        if not system_info:
+            self.logger.error('No particle information found in H5MD file.')
+            return
+
+        self._system_time_map = {}
+        for i_step, step in enumerate(self.trajectory_steps):
+            time = system_info[step].pop('time')
+            atoms_dict = system_info[step]
+
+            topology = None
+            if i_step == 0: # TODO extend to time-dependent bond lists and topologies
+                atoms_dict['bond_list'] = self._data_parser.get('connectivity.bonds')
+                path_topology = 'connectivity.particles_group'
+                topology = self._data_parser.get(path_topology)
+
+            self.parse_trajectory_step({
+                'atoms': atoms_dict
+            })
+
+            if i_step == 0 and topology: # TODO extend to time-dependent topologies
+                self.parse_atomsgroup(sec_run.system[i_step], topology, path_topology)
+
+    def parse_method(self):
+
+        sec_method = Method()
+        self.archive.run[-1].method.append(sec_method)
+        sec_force_field = sec_method.m_create(ForceField)  # TODO @landinesa how do we remove m_create's here?
```

Comment:
yes please avoid using m_create. The reason being that since we are moving in the direction of base classes, m_create would not work in case the parent class was used to define a sub_section.

Reply:
yes, I wanted to change this, but I cannot figure out the correct approach in this case:

```
        sec_force_field = ForceField()
        self.archive.run[-1].method[-1].force_field.append(sec_force_field)
```
AttributeError: 'NoneType' object has no attribute 'append'...but I don't understand because in my mind this should be the same as how we treat the run section earlier in the code??

Reply:
ok, sorry, that was stupid. I figured this out:
```
  sec_method = Method()
  self.archive.run[-1].method.append(sec_method)
  sec_force_field = ForceField()
  sec_method.force_field = sec_force_field
  sec_model = Model()
  sec_force_field.model.append(sec_model)
```
Is this sufficient?

Reply:
it is really not obvious that repeating sections default to an empty list, otherwise None. yes that is correct.

Reply:
done.